src/coreclr/vm/jitinterface.cpp

```diff
@@ -11336,6 +11343,15 @@ WORD CEEJitInfo::getRelocTypeHint(void * target)
     }
 #endif // TARGET_AMD64
 
+#ifdef TARGET_ARM64
+    // Unlike x86/x86_64 we can't rely on IsPreferredExecutableRange here, instead we'll try to
+    // guess whether the native instruction (e.g. br) will be able to reach this target or not
+    // within 128Mb (1Mb is subtracted because we don't know the exact code size at this point)
+    // NOTE: it's still just a hint
+    if (FitsInRel28(abs((INT64)m_CodeHeader->GetCodeStartAddress()) - (INT64)target) + 1024 * 1024)
```

Comment:
Ok, it's not correct to use `m_CodeHeader->GetCodeStartAddress()` here.
What I need here is an approximate location where currently jitted method will be allocated in memory, is it possible?

Or at very least I need a location of coreclr in memory so I can return IMAGE_REL_ARM64_BRANCH26 for everything except coreclr and all calls to coreclr will use direct calls.

Reply:
> What I need here is an approximate location where currently jitted method will be allocated in memory, is it possible?

It is certainly possible, but it is not a one line change with the current implementation of the code manager. I think we would need to refactor the implementation of code manager and how the JIT interacts with it to make this possible.

Reply:
I assume x64 will also benefit from it, the current `getRelocTypeHint` impl is not correct for it either and most likely leads to jump-stubs or "rejit with relocs off" in some cases.